mcs/class/corlib/System.Reflection.Emit/DynamicMethod.cs

```diff
@@ -133,6 +133,8 @@ public DynamicMethod (string name, Type returnType, Type[] parameterTypes, bool
 		[MethodImplAttribute(MethodImplOptions.InternalCall)]
 		private static extern void create_dynamic_method (DynamicMethod m);
 
+		// Clearing of ilgen in create_dynamic_method is not yet synchronized for multiple threads
+		[MethodImpl(MethodImplOptions.Synchronized)]
 		private void CreateDynMethod () {
 			if (mhandle.Value == IntPtr.Zero) {
```

Comment:
Would prefer an explicit lock, synchronized methods are complicated for the runtime to handle.